src/osd/ExtentCache.h

```diff
@@ -297,6 +304,19 @@ class ExtentCache {
 		extoff, extlen);
 	    }
 	    final_extent->link(*this, pin);
+	  } else if (action.action == update_action::RESERVE_EXT) {
+	    bufferlist bl;
+            bl.substr_of(
+                *(ext->bl),
```

Comment:
I *think* you're right that bl here must be populated because there can't be an in-progress write here, but you should add a comment explaining the invariant.